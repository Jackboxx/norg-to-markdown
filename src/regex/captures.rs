use regex::Regex;

pub fn lenght_of_nth_capture(regex: &Regex, line: &str, n: usize) -> usize {
    match regex.captures(line) {
        Some(cap) => match cap.get(n) {
            Some(mat) => mat.as_str().len(),
            None => 0,
        },
        None => 0,
    }
}
