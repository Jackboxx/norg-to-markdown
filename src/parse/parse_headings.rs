use regex::Regex;

use crate::regex::captures::lenght_of_nth_capture;

pub fn parse_headings(line: &str) -> Result<String, regex::Error> {
    let regex = Regex::new(r"(?m:^\*{1,5}\s)")?;
    let amount = lenght_of_nth_capture(&regex, line, 0);

    let arr = vec!["#"; amount];
    Ok(regex.replace(line, arr.join("") + " ").to_string())
}

#[cfg(test)]
mod tests {
    use super::parse_headings;

    #[test]
    fn test_parse_headings() {
        assert_eq!(parse_headings("text"), Ok("text".to_owned()));
        assert_eq!(parse_headings("* heading1"), Ok("# heading1".to_owned()));
        assert_eq!(parse_headings("*heading2"), Ok("*heading2".to_owned()));
        assert_eq!(
            parse_headings("** heading large"),
            Ok("## heading large".to_owned())
        );
        assert_eq!(
            parse_headings("not a * heading"),
            Ok("not a * heading".to_owned())
        );
    }
}
