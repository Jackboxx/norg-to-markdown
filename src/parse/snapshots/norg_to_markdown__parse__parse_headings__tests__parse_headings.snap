---
source: src/parse/parse_headings.rs
expression: res
---

# heading
- bullet point 1
- bullet point 2
- bullet point 3
## to do 
nothing
